addons/sale_timesheet/migrations/11.0.1.0/post-migration.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2018 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+from psycopg2.extensions import AsIs
+
+
+def migrate_sale_line_tasks(env):
+    openupgrade.logged_query(
+        env.cr, """
+            UPDATE sale_order_line sol
+            SET task_id = po.task_id
+            FROM procurement_order po
+            WHERE sol.id = po.sale_line_id
+        """)
+
+
+def migrate_project_sale_line(env):
+    openupgrade.logged_query(
+        env.cr, """
+            UPDATE project_project pj
+            SET sale_line_id = sol.id
+            FROM sale_order_line sol, product_product pd, sale_order so
+            WHERE sol.product_id = pd.id
+            AND sol.order_id = so.id
+            AND so.analytic_account_id = pj.id
```

Comment:
There are several things more to have here into account: the SO state, as a draft SO can't generate that project, and the product old tracking values, being sure that the product was configured to generate a new project.